artifact/v3/manifest/manifest.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <artifact/v3/manifest/manifest.hpp>
+
+#include <string>
+#include <sstream>
+
+#include <common/error.hpp>
+
+namespace mender {
+namespace artifact {
+namespace v3 {
+namespace manifest {
+
+namespace io = mender::common::io;
+namespace error = mender::common::error;
+
+const size_t expected_shasum_length {64};
+const size_t expected_whitespace {2};
+
+ExpectedManifest Parse(mender::common::io::Reader &reader) {
+	Manifest m {};
+
+	stringstream ss {};
+
+	io::StreamWriter sw {ss};
+
+	auto err = io::Copy(sw, reader);
+	if (err) {
+		return expected::unexpected(err);
+	}
+
+	string entry_name {};
+	string shasum {};
+	string line {};
```

Comment:
All of these can be declared inside the loop, ideally in the place where they are used.

Reply:
`Variables needed for if, while and for statements should normally be declared within those statements, so that such variables are confined to those scopes. E.g.:`

Seems ur right: https://google.github.io/styleguide/cppguide.html#Variable_and_Array_Initialization